src/reflect/scala/reflect/internal/Depth.scala

```diff
@@ -37,4 +37,9 @@ object Depth {
     if (depth < AnyDepthValue) AnyDepth
     else new Depth(depth)
   }
+
+  def maximumBy[A](xs: List[A], ff: A => Depth): Depth = {
```

Comment:
Functions with value classes will incur boxing/unboxing. Introducing a SAM type might be better:

```scala
trait DepthFunction[A] { def apply(a: A): Depth }
object Depth {
def maximumBy(xs: List[A])(ff: DepthFunction) = {
  var d = 0
  var rest = xs
  while (!rest.isEmpty) {
    d = ...
    rest = rest.tail
  }
}
```